wpengine/atlas-content-modeler
# [BH-1139] Show message and delete relationship references during model deletion

- Adds a warning about relationship fields and their data being removed in the Delete Model modal. The warning only appears if relationship fields exist that refer to the model about to be deleted.
- When a model is deleted:
   - Delete relationship fields that refer to the deleted model.
   - Delete references to entries of the deleted model from the post-to-post table.

https://wpengine.atlassian.net/browse/BH-1139

## Testing

### To test manually
- Create two models.
- Add a relationship field so that one model refers to the other.
- Add an entry and fill in the relationship field.
- Take note of the added row in the`wp_acm_post_to_post` table.
- Delete the model that the reference field refers to from the “Content Models” index.

Confirm that:
- You saw a red warning in the Delete Model modal about relationship fields being deleted.
- The relationship field was removed from the other model upon model deletion.
- The entry reference(s) have been removed from the `wp_acm_post_to_post` table upon deletion.

The PR also includes these tests:

### PHP
- Adds missing tests for the model deletion REST endpoint.
- Extends model deletion tests to verify new relationship cleanup functions.

### JS unit tests
- Jest tests for the new `getRelationships` function.

### e2e tests
- Adds e2e tests to check the warning message appears and that relationship fields are removed without a page refresh.

## Screenshots

<img width="608" alt="Screenshot 2021-09-06 at 19 00 07" src="https://user-images.githubusercontent.com/647669/132247221-679d01d6-f035-4766-b0ba-988639a17d09.png">

## Documentation Changes

In progress and covered separately by [BH-1216](https://wpengine.atlassian.net/browse/BH-1216).

## Dependant PRs

#250 should be merged first. This PR may need to be adjusted once that is merged, in particular the `RestModelEndpointTests`, because the field testing data has changed shape there.
